server/game.py

```diff
@@ -21,6 +21,29 @@ def _configure(max_game_time, agent_dir):
     MAX_GAME_TIME = max_game_time
     AGENT_DIR = agent_dir
 
+def fix_bc_path(path):
+    """
+    Loading a PPO agent trained with a BC agent requires loading the BC model as well when restoring the trainer, even though the BC model is not used in game
+    For now the solution is to include the saved BC model and fix the relative path to the model in the config.pkl file
+    """
+
+    import pickle 
+    #the path is the agents/Rllib.*/agent directory
+    agent_path = os.path.dirname(path)
+    with open(os.path.join(agent_path,"config.pkl"), "rb") as f:
+        data = pickle.load(f)
+    try:
+        bc_model_dir = data["bc_params"]["bc_config"]["model_dir"]
+        last_dir = os.path.basename(bc_model_dir)
+        bc_model_dir = os.path.join(agent_path,"bc_params",last_dir)
+        data["bc_params"]["bc_config"]["model_dir"]= bc_model_dir
+        with open(os.path.join(agent_path,"config.pkl"), "wb") as f:
+            pickle.dump(data,f)
+    except:
+        pass 
```

Comment:
Might be good to have better erroring here? Why are you using a try/except?

Reply:
So when I was first writing this I wasn't sure if the config file from trainer with sp agents would be the same as the one with ppo+bc agent, so just to be safe I put it in the try-except block. If an exception happens, it would be a key-error from trying to index into a dictionary with a wrong key. In that case it is probably a trainer with SP agents so it would be fine to just keep the config file the same. In hindsight the config should be the same and this is probably not necessary.